tasks/build-nagios.yml

```diff
@@ -6,6 +6,49 @@
 - name: Unpack Nagios source files
   shell: cd {{ download_dir }} && tar -xzvf {{ nagiossrc }}.tar.gz creates={{ download_dir }}/{{ nagiossrc }}
 
+- name: Set upgrade as not required by default
+  set_fact:
+    nagios_upgrade_required: false
+
+# https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/4/en/upgrading.html
+- name: check if nagios main.php exists
+  stat: path=/usr/local/nagios/share/main.php
+  register: nagios_mainphp_file
+
+- block: 
+
+  - name: read nagios installed version
+    shell: head /usr/local/nagios/share/main.php | grep "this_version" | cut -d"=" -f 2 | tr --delete ";" | tr --delete "'" | xargs echo -n
+    changed_when: false
+    register: cmd_nagios_version
+    tags:
+      - skip_ansible_lint
+    check_mode: no
+  
+  - name: Check if upgrade is required
+    set_fact:
+      nagios_upgrade_required: true
+    when: cmd_nagios_version.stdout != nagios_version
+  
+  when: nagios_mainphp_file.stat.exists
+
+- name: Clean files before upgrade
```

Comment:
I'm not sure about this task as it would delete files without warning and there's no way of knowing if anyone has made any custom changes that they want to keep in those files.